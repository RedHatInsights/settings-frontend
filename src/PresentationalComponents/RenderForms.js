import React from 'react';
import PropTypes from 'prop-types';
import { formFieldsMapper, layoutMapper } from '@data-driven-forms/pf4-component-mapper';
import { Skeleton } from '@redhat-cloud-services/frontend-components';
import { Stack, StackItem, Card, CardBody } from '@patternfly/react-core';
import FormRender from '@data-driven-forms/react-form-renderer';

const RenderForms = ({ schemas, loaded, saveValues, ...props }) => (
    <Stack { ...props }>
        { loaded
            ?  schemas.map((schema, i) => (
                <StackItem key={ `settings-form-${i}` }>
                    <Card>
                        <CardBody>
                            <FormRender
                                formFieldsMapper={ formFieldsMapper }
                                layoutMapper={ layoutMapper }
                                schema={ schema }
                                onSubmit={ saveValues }
                                canReset
                            />
                        </CardBody>
                    </Card>
                </StackItem>
            ))
            : <StackItem>
                <Card>
                    <CardBody>
                        <Skeleton size='lg' />
                    </CardBody>
                </Card>
            </StackItem>
        }
    </Stack>
);

RenderForms.propTypes = {
    schemas: PropTypes.arrayOf(PropTypes.shape({})),
    loaded: PropTypes.bool,
    appId: PropTypes.string,
    saveValues: PropTypes.func
};

RenderForms.defaultProps = {
    saveValues: () => undefined
};

export default RenderForms;
