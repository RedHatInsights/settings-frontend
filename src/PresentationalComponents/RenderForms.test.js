import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RenderForms from './RenderForms';

describe('component', () => {
  it('should render - no data', () => {
    const wrapper = shallow(<RenderForms />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render - no schema', () => {
    const wrapper = shallow(<RenderForms loaded schemas={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render - one schema', () => {
    const wrapper = shallow(<RenderForms loaded schemas={[{}]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render - multiple schema', () => {
    const wrapper = shallow(<RenderForms loaded schemas={[{}, {}]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
