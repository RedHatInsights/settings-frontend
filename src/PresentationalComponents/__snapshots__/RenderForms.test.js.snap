// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should render - multiple schema 1`] = `
<Stack
  className=""
  component="div"
  gutter={null}
>
  <StackItem
    className=""
    gutter="md"
    isFilled={false}
    key="settings-form-0"
  >
    <Card
      className=""
      component="article"
      isHoverable={false}
    >
      <CardBody
        className=""
        component="div"
        isFilled={true}
      >
        <FormRenderer
          buttonClassName=""
          buttonsLabels={Object {}}
          canReset={true}
          clearOnUnmount={false}
          disableSubmit={Array []}
          formFieldsMapper={
            Object {
              "checkbox": [Function],
              "date-picker": [Function],
              "field-array": [Function],
              "plain-text": [Function],
              "radio": [Function],
              "select-field": [Function],
              "sub-form": [Function],
              "switch-field": [Function],
              "tabs": [Function],
              "text-field": [Function],
              "textarea-field": [Function],
              "time-picker": [Function],
              "wizard": [Function],
            }
          }
          formType="pf3"
          initialValues={Object {}}
          layoutMapper={
            Object {
              "Button": [Function],
              "ButtonGroup": [Function],
              "Description": [Function],
              "FormWrapper": [Function],
              "Title": [Function],
            }
          }
          onSubmit={[Function]}
          resetAble={false}
          schema={Object {}}
          schemaType="default"
          showFormControls={true}
          uiSchema={Object {}}
        />
      </CardBody>
    </Card>
  </StackItem>
  <StackItem
    className=""
    gutter="md"
    isFilled={false}
    key="settings-form-1"
  >
    <Card
      className=""
      component="article"
      isHoverable={false}
    >
      <CardBody
        className=""
        component="div"
        isFilled={true}
      >
        <FormRenderer
          buttonClassName=""
          buttonsLabels={Object {}}
          canReset={true}
          clearOnUnmount={false}
          disableSubmit={Array []}
          formFieldsMapper={
            Object {
              "checkbox": [Function],
              "date-picker": [Function],
              "field-array": [Function],
              "plain-text": [Function],
              "radio": [Function],
              "select-field": [Function],
              "sub-form": [Function],
              "switch-field": [Function],
              "tabs": [Function],
              "text-field": [Function],
              "textarea-field": [Function],
              "time-picker": [Function],
              "wizard": [Function],
            }
          }
          formType="pf3"
          initialValues={Object {}}
          layoutMapper={
            Object {
              "Button": [Function],
              "ButtonGroup": [Function],
              "Description": [Function],
              "FormWrapper": [Function],
              "Title": [Function],
            }
          }
          onSubmit={[Function]}
          resetAble={false}
          schema={Object {}}
          schemaType="default"
          showFormControls={true}
          uiSchema={Object {}}
        />
      </CardBody>
    </Card>
  </StackItem>
</Stack>
`;

exports[`component should render - no data 1`] = `
<Stack
  className=""
  component="div"
  gutter={null}
>
  <StackItem
    className=""
    isFilled={false}
  >
    <Card
      className=""
      component="article"
      isHoverable={false}
    >
      <CardBody
        className=""
        component="div"
        isFilled={true}
      >
        <Skeleton
          size="lg"
        />
      </CardBody>
    </Card>
  </StackItem>
</Stack>
`;

exports[`component should render - no schema 1`] = `
<Stack
  className=""
  component="div"
  gutter={null}
/>
`;

exports[`component should render - one schema 1`] = `
<Stack
  className=""
  component="div"
  gutter={null}
>
  <StackItem
    className=""
    gutter="md"
    isFilled={false}
    key="settings-form-0"
  >
    <Card
      className=""
      component="article"
      isHoverable={false}
    >
      <CardBody
        className=""
        component="div"
        isFilled={true}
      >
        <FormRenderer
          buttonClassName=""
          buttonsLabels={Object {}}
          canReset={true}
          clearOnUnmount={false}
          disableSubmit={Array []}
          formFieldsMapper={
            Object {
              "checkbox": [Function],
              "date-picker": [Function],
              "field-array": [Function],
              "plain-text": [Function],
              "radio": [Function],
              "select-field": [Function],
              "sub-form": [Function],
              "switch-field": [Function],
              "tabs": [Function],
              "text-field": [Function],
              "textarea-field": [Function],
              "time-picker": [Function],
              "wizard": [Function],
            }
          }
          formType="pf3"
          initialValues={Object {}}
          layoutMapper={
            Object {
              "Button": [Function],
              "ButtonGroup": [Function],
              "Description": [Function],
              "FormWrapper": [Function],
              "Title": [Function],
            }
          }
          onSubmit={[Function]}
          resetAble={false}
          schema={Object {}}
          schemaType="default"
          showFormControls={true}
          uiSchema={Object {}}
        />
      </CardBody>
    </Card>
  </StackItem>
</Stack>
`;
